rfd/0029-account-lifecycle.md

```diff
@@ -0,0 +1,231 @@
+a recovery token:
+
+* Login with a username and password.
+* Instead of pressing second factor, enter a one-time recovery token
+* Enroll new second factor.
+* All recovery tokens are no longer valid. The new set of recovery tokens are printed to the user.
+* User has to confirm that they have copied recovery tokens.
+* The session is not initiated. A user is redirected to a login form
+with UI displaying a message to login using new credentials.
+
+* If the operation is aborted after user enters the token, but fails to enter the password,
+the valid token is no longer used, but remaining tokens can be used.
+* If a user tries to reuse the token, 
+
+This flow ensures two factors are verified: the valid password and a recovery token.
+
+**User lost a password**
+
+If user lost a password, but not a second factor, they could recover using
+a recovery link sent to their email and a second factor:
+
+* User enters their username and clicks forgot password.
+* User clicks have to present a valid second factor and a recovery token,
+after that they will be able to reset their password.
+* All recovery tokens are no longer valid. The new set of recovery tokens are printed to the user.
+* User has to confirm that they have copied recovery tokens.
+* The session is not initiated. A user is redirected to a login form
+with UI displaying a message to login using new credentials.
+
+This flow ensure that two factors are verified:
+
+* Second factor TOTP token or U2F device.
+* A valid account recovery token.
+
+**Getting new tokens**
+
+If any user is logged into Teleport UI, they can regenerate new password recovery tokens:
+
+* Alice goes to her account settings and chooses to generate new set of recovery tokens.
+* Teleport asks for a second factor before generating tokens and makes it clear that once
+selected, old sets of tokens will not be valid, they will be printed once and user
+should be prepared to store them in offline place.
+* A new set of tokens is printed for Alice. The message makes it clear
+that once closed, the tokens are no longer going to be revealed and that any previous
+tokens are invalid.
+
+**Internal request for canceling the account**
+
+To request account cancellation, any account owner should:
+
+* Be logged into Teleport UI.
+* Press a "cancel my account" button in the UI.
+* UI should verify that they own a second factor.
+* Verify that they have write permission on the billing.
+
+```
+We have marked your account as inactive. In 10 days, we are going to delete the account and all it's data.
+To prevent this, log into Teleport and press "revert account cancellation".
+```
+
+The account enters inactive state:
+
+**Inactive account state**
+
+* Teleport should not disconnect any resources connected to the cloud account during the grace period.
+Otherwise it is possible for attacker overtaking an account owner to cause system wide outage.
+* Account owner will be able to login in the account, but will
```

Comment:
```suggestion
* Users will be able to login in the account, but will
```

Reply:
I've updated to lock all non-account owner users out of inactive account.